volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,306 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import datetime
+from typing import Iterable, List
+
+from volatility.framework import constants, exceptions, interfaces, renderers, symbols, layers
+from volatility.framework.automagic.pdbscan import scan
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.symbols import intermed
+from volatility.framework.symbols.windows.extensions import network
+from volatility.plugins import timeliner
+from volatility.plugins.windows import poolscanner
+
+vollog = logging.getLogger(__name__)
+
+class NetScan(interfaces.plugins.PluginInterface):
+    """Scans for network objects present in a particular windows memory image."""
+
+    _version = (1, 0, 0)
+    CORRUPT_DEFAULT = False
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.PluginRequirement(name = 'poolscanner', plugin = poolscanner.PoolScanner, version = (1, 0, 0)),
+            requirements.BooleanRequirement(name = 'include-corrupt',
```

Comment:
For now I've included a flag which shows objects which fulfill the Poolheader requirement but little more. This is because vol2 sometimes found alleged endpoints which only fulfilled the _structural_ requirements without actually having valid _content_. This could actually lead to things like showing what looks like a valid IP address, but which in reality was a pointer being parsed as an IP address.
`--include-corrupt` simulates this behaviour and right now literally only exists because it is possible. I am *not* sure if I should keep this option because I severely doubt it will be useful, but it might be in ony tiny edge-case scenario...

Reply:
The flag doesn't overly hurt, but I'm also happy for it to be dropped if it really doesn't serve a purpose.

Reply:
Unless someone has a reasonable argument against it I would like to keep it for now. Removing it is easily done if need arises.

Reply:
Yep, that'll be fine.